# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1498800091 14400
#      Fri Jun 30 01:21:31 2017 -0400
# Node ID ab9e51f41a29db47e76badfb13436880e293a170
# Parent  b2d55b52ee5173b7b0b122e89036082881a2bfc8
substitute no-edit warning in oct-errno.in.cc

* oct-errno.in.cc: Use substitution for no-edit warning.

* mk-errno-list.sh: Handle substitution with sed here instead of in
module.mk.  Substitute no-edit warning.

* module.mk: Update command for oct-errno.cc

diff --git a/libinterp/corefcn/mk-errno-list.sh b/libinterp/corefcn/mk-errno-list.sh
--- a/libinterp/corefcn/mk-errno-list.sh
+++ b/libinterp/corefcn/mk-errno-list.sh
@@ -14,39 +14,48 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 if [ $# -ne 2 ]; then
-  echo "usage: get-errno-list [--perl PERL|--python PYTHON]" 1>&2
+  echo "usage: get-errno-list [--perl PERL|--python PYTHON|--sed SED]" 1>&2
   exit 1
 fi
 
 if [ $1 = "--perl" ]; then
   PERL="$2"
   $PERL -e 'foreach $key (sort (keys (%!))) {
     $x .= "#if defined ($key)\n    { \"$key\", $key, },\n#endif\n";
   }
   while (<>) {
-    s/^ *\@SYSDEP_ERRNO_LIST\@/$ x/;
+    s/^ *\@SYSDEP_ERRNO_LIST\@/$x/;
+    s/\@NO_EDIT_WARNING\@/DO NOT EDIT!  Generated automatically from oct-errno.in.cc by mk-errno-list.sh/;
     print;
   }'
 
 elif [ $1 = "--python" ]; then
   PYTHON="$2"
   $PYTHON -c '
 from errno import errorcode
 from sys import stdin, stdout
 
 t = "#if defined (%s)\n    { \"%s\", %s, },\n#endif\n"
 errstr = ""
 for v in sorted (errorcode.values ()):
     errstr += t % tuple (3 * [v])
 
+noedit = "DO NOT EDIT!  Generated automatically from oct-errno.in.cc by mk-errno-list.sh"
+
+repls = ("@SYSDEP_ERRNO_LIST@", errstr), ("@NO_EDIT_WARNING@", noedit)
+
 for l in stdin:
-    stdout.write (l.replace ("@SYSDEP_ERRNO_LIST@", errstr))
+  stdout.write (reduce (lambda a, kv: a.replace (*kv), repls, l))
 '
+elif [ $1 = "--sed" ]; then
+  SED="$2"
+  $SED -e '/@SYSDEP_ERRNO_LIST@/D' \
+       -e 's/@NO_EDIT_WARNING@/DO NOT EDIT!  Generated automatically from oct-errno.in.cc by mk-errno-list.sh/'
 fi
 
 exit $?
diff --git a/libinterp/corefcn/module.mk b/libinterp/corefcn/module.mk
--- a/libinterp/corefcn/module.mk
+++ b/libinterp/corefcn/module.mk
@@ -267,17 +267,17 @@ COREFCN_SRC = \
 
 %reldir%/oct-errno.cc: %reldir%/oct-errno.in.cc %reldir%/mk-errno-list.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
 	if test -n "$(PERL)"; then \
 	  $(SHELL) $(srcdir)/%reldir%/mk-errno-list.sh --perl "$(PERL)" < $< > $@-t; \
 	elif test -n "$(PYTHON)"; then \
 	  $(SHELL) $(srcdir)/%reldir%/mk-errno-list.sh --python "$(PYTHON)" < $< > $@-t; \
 	else \
-	  $(SED) '/@SYSDEP_ERRNO_LIST@/D' $< > $@-t; \
+	  $(SHELL) $(srcdir)/%reldir%/mk-errno-list.sh --sed "$(SED)" < $< > $@-t; \
 	fi && \
 	mv $@-t $@
 
 %reldir%/mxarray.h: %reldir%/mxarray.in.h %reldir%/mk-mxarray-h.sh | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)$(call simple-filter-rule,%reldir%/mk-mxarray-h.sh)
 
 %reldir%/oct-tex-lexer.ll: %reldir%/oct-tex-lexer.in.ll %reldir%/oct-tex-symbols.in | %reldir%/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
diff --git a/libinterp/corefcn/oct-errno.in.cc b/libinterp/corefcn/oct-errno.in.cc
--- a/libinterp/corefcn/oct-errno.in.cc
+++ b/libinterp/corefcn/oct-errno.in.cc
@@ -1,9 +1,9 @@
-// DO NOT EDIT!  Generated automatically from oct-errno.in.cc by configure
+// @NO_EDIT_WARNING@
 /*
 
 Copyright (C) 2005-2017 John W. Eaton
 
 This file is part of Octave.
 
 Octave is free software; you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by
